src/commonAccountFunctions.ts

```diff
@@ -109,6 +111,78 @@ export async function getQuery<T extends Account>(this: T, name: string) {
   })) as Models.Query;
   return new Query(app, query, this);
 }
+export interface RunPipelineOpts {
+  onProgress?: (progress: PipelineProgress) => void;
+  /**
+   * Execute these queries in parallel
+   */
+  queries: Array<Query | { query: Query; priority?: number }>;
+  /**
+   * Store the results in this dataset
+   */
+  destination: {
+    dataset: Dataset;
+    graph?: Graph | string;
+  };
+  /**
+   * Execute the queries on this dataset. If all queries use the same dataset, you can leave this unset
+   */
+  source?: Dataset;
+}
+export async function runPipeline<T extends Account>(this: T, opts: RunPipelineOpts) {
+  if (!opts.queries.length) throw getErr("No queries given to run pipeline on");
+  const queries: Array<{ query: Query; priority?: number }> = opts.queries.map((q) => {
+    if (q instanceof Query) {
+      return {
+        query: q,
+        priority: undefined,
+      };
+    } else {
+      return q;
+    }
+  });
+  let sourceDataset: Dataset | undefined = opts.source;
+  if (!sourceDataset) {
+    const queryInfos = await Promise.all(queries.map((q) => q.query.getInfo()));
+    const usedDatasets = uniq(
+      compact(
+        queryInfos.map((info) => {
+          if (!info.dataset) return undefined;
+          return `${info.dataset.owner.accountName}/${info.dataset.name}`;
+        }),
+      ),
+    );
+    if (usedDatasets.length !== 1) {
+      throw getErr(`Cannot create pipeline: pass a source document first`);
+    }
+    // Cast to dataset, as we know we can populate this field
+    // (otherwise, we wouldnt have had metadata info in the original pipeline object about this dataset)
+    sourceDataset = (await queries[0].query.getDataset()) as Dataset;
```

Comment:
This doesn't work: the first query could have no dataset defined, while a later query does have a dataset.

Maybe you can create an object instead of using `uniq`? Then you can use the slug (I thought there was a `.slug` property now for datasets too?) to check for uniqueness but still keep the `Dataset` objects around.